server.js

```diff
@@ -552,15 +837,15 @@ function setRoles() {
     console.log("shPlayers = " + shPlayers);
     console.log("numFascists = " + numFascists);
     console.log("numLiberals = " + numLiberals);
-
+    autoChancellor();
     return roles;
 
 }
 
 function nextPresident() {
     var index = presIndex;
     prevPresNom = presNom;
-    if (index == shPlayers.length - 1)
+    if (index >= shPlayers.length - 1)
```

Comment:
fix for "president is unidentified"
this occurs when the previous president is the last in the array (for 5 people, index = 4) and someone is executed. shPlayers.length then equals the index (4 in this case) and index is then greater than shPlayerslength - 1.  presNom is then shPlayers[index+1] (5 in this case) which doesn't exist.